### PR TITLE
Use BraveTabHoverCardController instead of upstream implementation

### PR DESCRIPTION
**Pull request by sangwoo108**
Resolves https://github.com/brave/brave-browser/issues/26251

This was accidentally missed during rebase.

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

#### Automated
Has compile time check that ensure we're using our own implementation

#### Manual
tab hover card with image should be rendered properly

### CODE REVIEW

File: chromium_src/chrome/browser/ui/ui_features.cc [hunk 1]
```diff
@@ -11,7 +11,7 @@
 namespace features {
 
 OVERRIDE_FEATURE_DEFAULT_STATES({{
-    {kTabHoverCardImages, base::FEATURE_DISABLED_BY_DEFAULT},
+    {kTabHoverCardImages, base::FEATURE_ENABLED_BY_DEFAULT},
```
Comment (sangwoo108) [inline chromium_src/chrome/browser/ui/ui_features.cc:14]: Doesn't seem to be a right fix.. still digging.

File: chromium_src/chrome/browser/ui/views/tabs/tab_strip.cc [hunk 2]
```diff
@@ -23,6 +23,7 @@
   AddTab(std::make_unique<BraveTab>(this), MODEL_INDEX, PINNED)
 #define CompoundTabContainer BraveCompoundTabContainer
 #define TabContainerImpl BraveTabContainer
+#define TabHoverCardController BraveTabHoverCardController
```
[the remark this answers is not in the document]
Comment (fallaciousreasoning) [reply chromium_src/chrome/browser/ui/views/tabs/tab_strip.cc:26]: :+1:

File: chromium_src/chrome/browser/ui/views/tabs/tab_strip.h [hunk 3]
```diff
@@ -19,14 +23,23 @@
...
   virtual TabDragContext* GetDragContext
+#define TabHoverCardController BraveTabHoverCardController
 #include "src/chrome/browser/ui/views/tabs/tab_strip.h"
+#undef TabHoverCardController
 #undef GetDragContext
 #undef ShouldDrawStrokes
 #undef UpdateHoverCard
 
+static_assert(TabStrip::IsUsingBraveTabHoverCardController(),
+              "Should use BraveTabHoverCardController");
```
[the remark this answers is not in the document]
Comment (fallaciousreasoning) [reply chromium_src/chrome/browser/ui/views/tabs/tab_strip.h:43]: :+1:

**Review by fallaciousreasoning — APPROVED** [review]
LGTM!

**Review by goodov — APPROVED** [review]

**Comment by kjozwiak** [comment]
Verification **PASSED** on `Win  11 x64` using the following build(s):

```
Brave | 1.47.24 Chromium: 107.0.5304.68 (Official Build) nightly (64-bit)
-- | --
Revision | a4e93e89d3b3df1be22214603fba846ad0183ca5-refs/branch-heads/5304@{#991}
OS | Windows 11 Version 22H2 (Build 22621.675)
```

**`Card`** | **`Card with Preview`** | **`Card with Tooltip`**
-----------|---------------------------|---------------------------
card1 | cardPreview1 | tooltipCard

**Pull request merged**